sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -271,6 +272,25 @@ export class AssessmentController {
           );
           return job.complete();
         }
+        const application = assessment.application;
+        if (application.applicationStatus === ApplicationStatus.Completed) {
+          const updatedLastReportedAssessment =
+            await this.studentAssessmentService.updateLastReportedAssessment(
+              job.variables.assessmentId,
+              application.id,
+              assessment.offering.studyStartDate,
+              assessment.offering.studyEndDate,
+            );
+          if (updatedLastReportedAssessment === undefined) {
+            jobLogger.log("No previous reported assessment found.");
+          } else if (!updatedLastReportedAssessment) {
+            jobLogger.log(
+              "Last reported assessment id was previously updated or need not be updated.",
+            );
+          }
+        } else {
+          jobLogger.log("Application is not yet completed.");
+        }
```

Comment:
I will strongly suggest to keep the response parsing more simple. 

1. Remove the application not yet completed log.
2. If update method return true, log last reported assessment was updated.
3. If update method return false, log last reported assessment not found or updated previously.

Reply:
@dheepak-aot I have made adjustments here. It is slightly different. Please let me know if it is ok.